aws/resource_aws_cloudfront_key_group_test.go

```diff
@@ -0,0 +1,282 @@
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_cloudfront_key_group", &resource.Sweeper{
+		Name: "aws_cloudfront_key_group",
+		F:    testSweepCloudFrontKeyGroup,
+	})
+}
+
+func testSweepCloudFrontKeyGroup(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("Error getting client: %w", err)
+	}
+	conn := client.(*AWSClient).cloudfrontconn
+	var sweeperErrs *multierror.Error
+
+	input := &cloudfront.ListKeyGroupsInput{}
+
+	for {
+		output, err := conn.ListKeyGroups(input)
+		if err != nil {
+			if testSweepSkipSweepError(err) {
+				log.Printf("[WARN] Skipping CloudFront key group sweep for %s: %s", region, err)
+				return nil
+			}
+			sweeperErrs = multierror.Append(sweeperErrs, fmt.Errorf("error retrieving CloudFront key group: %w", err))
+			return sweeperErrs.ErrorOrNil()
+		}
+
+		if output == nil || output.KeyGroupList == nil || len(output.KeyGroupList.Items) == 0 {
+			log.Print("[DEBUG] No CloudFront key group to sweep")
+			return nil
+		}
+
+		for _, item := range output.KeyGroupList.Items {
+			strId := aws.StringValue(item.KeyGroup.Id)
+			log.Printf("[INFO] CloudFront key group %s", strId)
+			_, err := conn.DeleteKeyGroup(&cloudfront.DeleteKeyGroupInput{
+				Id: item.KeyGroup.Id,
+			})
+			if err != nil {
+				sweeperErr := fmt.Errorf("error deleting CloudFront key group %s: %w", strId, err)
+				log.Printf("[ERROR] %s", sweeperErr)
+				sweeperErrs = multierror.Append(sweeperErrs, sweeperErr)
+				continue
+			}
+		}
+
+		if output.KeyGroupList.NextMarker == nil {
+			break
+		}
+		input.Marker = output.KeyGroupList.NextMarker
+	}
+
+	return sweeperErrs.ErrorOrNil()
+}
+
+func TestAccAWSCloudFrontKeyGroup_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	resourceName := "aws_cloudfront_key_group.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(cloudfront.EndpointsID, t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudFrontKeyGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudFrontKeyGroupConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudFrontKeyGroupExistence(resourceName),
+					resource.TestCheckResourceAttr("aws_cloudfront_key_group.test", "comment", "test key group"),
+					resource.TestCheckResourceAttrSet("aws_cloudfront_key_group.test", "etag"),
+					resource.TestCheckResourceAttrSet("aws_cloudfront_key_group.test", "id"),
+					resource.TestCheckResourceAttr("aws_cloudfront_key_group.test", "items.#", "1"),
+					resource.TestCheckResourceAttr("aws_cloudfront_key_group.test", "name", fmt.Sprintf("tf-acc-test-%d", rInt)),
```

Comment:
Nit: Rather than passing a random integer and reconstructing a full name in the test configuration and in the test checks, it is typically easier to construct a random name via `rName := acctest.RandomWithPrefix("tf-acc-test")` and use that in both locations. 👍